packages/odf/components/s3-browser/bucket-policy/BucketPolicy.tsx

```diff
@@ -0,0 +1,373 @@
+    });
+
+  return (
+    <>
+      {noPolicyExists && (
+        <span className="pf-v5-u-mt-sm">
+          <Button
+            variant={ButtonVariant.primary}
+            onClick={launchSaveModal}
+            className="pf-v5-u-mr-xs"
+          >
+            {t('Apply policy')}
+          </Button>
+          <Button
+            variant={ButtonVariant.link}
+            onClick={() => setCode('')}
+            className="pf-v5-u-ml-xs"
+          >
+            {t('Clear')}
+          </Button>
+        </span>
+      )}
+      {!noPolicyExists && (
+        <span className="pf-v5-u-mt-sm">
+          <Button
+            variant={ButtonVariant.secondary}
+            onClick={launchSaveModal}
+            className="pf-v5-u-mr-xs"
+          >
+            {t('Save changes')}
+          </Button>
+          <Button
+            variant={ButtonVariant.secondary}
+            onClick={launchDeleteModal}
+            className="pf-v5-u-mr-xs pf-v5-u-ml-xs"
+            isDanger
+          >
+            {t('Delete')}
+          </Button>
+          <Button
+            variant={ButtonVariant.link}
+            onClick={triggerRefresh}
+            className="pf-v5-u-ml-xs"
+          >
+            {t('Cancel')}
+          </Button>
+        </span>
+      )}
+    </>
+  );
+};
+
+const BucketPolicyContent: React.FC<BucketPolicyContentProps> = ({
+  success,
+  setSuccess,
+  triggerRefresh,
+}) => {
+  const [code, setCode] = React.useState('');
+  const [edit, setEdit] = React.useState(false);
+
+  const { noobaaS3 } = React.useContext(NoobaaS3Context);
+  const { bucketName } = useParams();
+  const {
+    data: policyData,
+    error,
+    isMutating: isLoading,
+    trigger,
+  } = useSWRMutation(`${bucketName}-${BUCKET_POLICY_CACHE_KEY_SUFFIX}`, () =>
+    noobaaS3.getBucketPolicy({ Bucket: bucketName })
+  );
+
+  const noPolicyExists =
+    error?.name === 'NoSuchBucketPolicy' && !policyData?.Policy;
+
+  // initial fetch on first mount or remounts only
+  React.useEffect(() => {
+    const fetchPolicies = async () => {
+      try {
+        await trigger();
+      } catch (err) {
+        // no need to handle any error here, use "error" object directly from the "useSWRMutation" hook
+        // eslint-disable-next-line no-console
+        console.error(err);
+      }
+    };
+    fetchPolicies();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  React.useEffect(() => {
+    if (!isLoading && !error && !!policyData?.Policy) {
+      const unformattedPolicy = policyData?.Policy;
+      try {
+        const formattedPolicy = JSON.stringify(
+          JSON.parse(unformattedPolicy),
+          null,
+          2
+        );
+        setCode(formattedPolicy);
+      } catch {
+        setCode(unformattedPolicy);
+      }
+    }
+  }, [isLoading, error, policyData]);
+
+  if (isLoading || (error && !noPolicyExists)) {
+    return <StatusBox loaded={!isLoading} loadError={isLoading ? '' : error} />;
```

Comment:
Is the loadError ternary operator really needed?
```suggestion
    return <StatusBox loaded={!isLoading} loadError={error} />;
```

Reply:
yeah it was needed earlier when I was using `useSWR` hook... but later I moved to `useSWRMutation` so not needed anymore. I will update, thanks !